zookeeper-docs/src/main/resources/markdown/zookeeperAdmin.md

```diff
@@ -1508,13 +1508,40 @@ and [SASL authentication for ZooKeeper](https://cwiki.apache.org/confluence/disp
     in such case, both Java and C client will close the session with server thereafter,
     without further attempts on retrying to reconnect.
 
+    This configuration is short hand for **enforce.auth.enabled=true** and **enforce.auth.scheme=sasl**
+
     By default, this feature is disabled. Users who would like to opt-in can enable the feature
-    by setting **zookeeper.sessionRequireClientSASLAuth** to **true**.
+    by setting **sessionRequireClientSASLAuth** to **true**.
 
     This feature overrules the <emphasis role="bold">zookeeper.allowSaslFailedClients</emphasis> option, so even if server is
     configured to allow clients that fail SASL authentication to login, client will not be able to
     establish a session with server if this feature is enabled.
 
+* *enforce.auth.enabled* :
+    (Java system property : **zookeeper.enforce.auth.enabled**)
+    **New in 3.7.0:**
+    When set to **true**, ZooKeeper server will only accept connections and requests from clients
+    that have authenticated with server via configured auth scheme. Authentication scheme
+    can be configured using property enforce.auth.scheme. Clients that are not
+    configured with the auth scheme configured at server or configured but failed authentication (i.e. with invalid credential)
+    will not be able to establish a session with server. A typed error code (-124) will be delivered
+    in such case, both Java and C client will close the session with server thereafter,
+    without further attempts on retrying to reconnect.
+
+    By default, this feature is disabled. Users who would like to opt-in can enable the feature
+    by setting **enforce.auth.enabled** to **true**.
+
+    When **enforce.auth.enabled=true** and **enforce.auth.scheme=sasl** then <emphasis role="bold">zookeeper.allowSaslFailedClients</emphasis> configuration is overruled. So even if server is
+    configured to allow clients that fail SASL authentication to login, client will not be able to
+    establish a session with server if this feature is enabled.
+
+* *enforce.auth.scheme* :
+    (Java system property : **zookeeper.enforce.auth.scheme**)
+    **New in 3.7.0:**
+    Specifies the authentication scheme with which all clients must get authenticated before doing
+    any zookeeper operations. This property is used only when **enforce.auth.enabled** is to
+    **true**.
```

Comment:
I am not sure it is a real case in production somewhere to have different auth scheme in the same cluster

do you have any case or possible scenario @ztzg  ?

Reply:
@ztzg I missed your comment above sorry

Reply:
+1 for @ztzg idea if he has a real usecase

Reply:
Yes, we have a real use-case.  (We would like to use SASL only, but it's going to take a looong time for everybody to transition.)

Reply:
I think it is better to add support for multiple authentication schemes. It does not cause any problem for one authentication scheme.  It will give more flexibility to user to handle all their possible use cases,

Reply:
Done